src/module.ts

```diff
@@ -2449,7 +2449,6 @@ export function readString(ptr: usize): string | null {
         }
       }
     }
-    arr.push(cp);
```

Comment:
@dcodeIO During play around with optimizing utf8 encoding / decoding I found this unnecessary push which cause to wrong decoding for large unicode points like `🚀`

Reply:
Also since 2003 UTF-8 has a [restriction](https://en.wikipedia.org/wiki/UTF-8#Description) which require 21-bit per value and 4 bytes instead 6.

Reply:
Can we make this a separate PR, so we get a fix release out of it?

Reply:
Yes, sure!